wordpress-mobile/WordPress-Android
# IA Reader add dedicated tracking for time spend in sub-filtered streams.

Fixes #11324

This is an addition to the already merged PR #10956 and adds the `time_in_subfiltered_list` property to the already available `application_closed` event.

This is a specific event of the IA revised reader (zalpha flavor) and reports the seconds spent in the main reader `FOLLOWING` tab when a `SITE` or `TAG` is selected in the sub-filter.

## To test
- Unit tests must pass
- Be sure you have tracking turned on (Me > App Settings > Privacy Settings > Collect information must be active)
- you can observe logcat while making following steps, filtering with this `(TRACK READER|Tracked: application_closed)` and Regex active
- Open the zalpha flavor of the app
- Go to Following and select a subfilter Site/Tag (stay there for something like 5 seconds); 
- Move to a different tab (like Discover) and stay there for other 5 seconds
- Open a post from Discover list and stay there for other 5 seconds 
- Come back to Following and wait 5 seconds
- Clear the filter and wait 5 seconds
- Put the app in background
- Check that you get some message like 
`Tracked: application_closed, Properties: {"time_in_main_reader":20,"last_visible_screen":"Reader","time_in_reader_paged_post":5,"time_in_app":34,"time_in_subfiltered_list":10,"time_in_reader_filtered_list":0}`
and the reported timings are somewhat similar to the expected
- Smoke test the app selecting other features different from reader and check the reader timing does not accumulate

PR submission checklist:
